### PR TITLE
Add for="" property to BooleanField labels

### LINKED ISSUE #658: Registration form: "I am over 16 years old" label not linked to checkbox
## Steps to reproduce the issue
0. Don't be logged in.
1. Go to https://accounts.fedoraproject.org/
2. Click on "I'm over 16 years old"

## Expected
Clicking the label toggles the checkbox. Also, screen readers know that this is the label for the checkbox.

## Actual
Clicking the label does nothing. Screen readers have to guess that this is the label for the checkbox.
The `<label>` is missing a `for` attribute with the value of the `<input>`'s `id`.

## What it looks like
Clicking "I am over 16 years old" doesn't toggle the checkbox.

## What it should look like
Clicking "I am over 16 years old" toggles the checkbox.

## Code in question
https://github.com/fedora-infra/noggin/blob/7357a1d3562dd3108de2ec17d6404387a7551a4d/noggin/form/register_user.py#L42-L43

This may be an upstream issue of wtforms and their [BooleanField](https://github.com/wtforms/wtforms/blob/a55be54c0b8487f81e8c0f61279818ada0bd2075/src/wtforms/fields/core.py#L849). However, I'm not familiar with either projects and couldn't find the code in question.

### PR DESCRIPTION
**Pull request by jaller94**
This is just a guess of what might needs to be done to fix #658.

Someone with a working development environment should verify this change.

### CODE REVIEW

File: noggin/templates/_form_macros.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
   {% endif %}
   {{ field(class=css_class, **kwargs) }}
   {% if field.type == 'BooleanField' %}
-      <label class="form-check-label ml-1"><strong>{{ field.label.text }}</strong></label>
+    <label class="form-check-label ml-1" {% if field.id %}for="{{ field.id }}"{% endif %}><strong>{{ field.label.text }}</strong></label>
```
Comment (abompard) [inline noggin/templates/_form_macros.html:18]: I don't think you need the test here, fields of forms built with WTForms always have an ID.

**Review by abompard — CHANGES_REQUESTED** [review]
Just one minor comment. Also, please use `-s` when committing to add a "signoff" line to the commit message.

Thanks!

**Comment by jaller94** [comment]
Hi @abompard, I don't have more time to spend on this.
I was just looking to make a reasonable, quick suggestion of how the issue I found might get fixed.

Feel free to take my change, re-commit or change it. No need for this PR/branch to get merged, if it's not compliant with the repo's license.

**Comment by abompard** [comment]
Sure, thanks @jaller94.

**Comment by abompard** [comment]
Superseded by 81d38e4.

**Pull request closed without merging**